### ISSUE TITLE
Ios build error with plugin added, fails on ci server / IonicPro

### ISSUE THREAD

**Issue by developerjohan01** [ionic-team/ionic-native-google-maps#81]
**I'm submitting a ...**  (check one with "x")
- [ ] question
- [x ] any problem or bug report
- [ ] feature request

**If you choose 'problem or bug report', please select OS:**  (check one with "x")
- [ ] **Android**
- [ x] **iOS**
- [ ] **Browser**

**cordova information:**  (run `$> cordova plugin list`) 

```
com.googlemaps.ios 2.7.0 "Google Maps SDK for iOS"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-googlemaps 2.4.4 "cordova-plugin-googlemaps"
cordova-plugin-ionic-keyboard 2.1.2 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 2.1.4 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
```

**If you use `@ionic-native/google-maps`, please tell the package.json (only `@ionic-native/core` and `@ionic-native/google-maps` are fine mostly)**
```
    "@ionic-native/core": "^4.14.0",
    "@ionic-native/google-maps": "^4.14.0",
```
**Current behavior:**
CI Server build/Ionic Pro is failing with the plugin installed, it worked previously. 

**Expected behavior:**
CI Server build/Ionic Pro should work with the plugin installed.

**Related code, data or error log (please format your code or data):**
My project that previously built without problem is now failing - also re-running old builds from previous commits that worked now fails.
I can recreate the problem locally, but the build really need to work on the server... 
The easiest way to recreate the problem is to use this sample project: https://github.com/developerjohan01/TabMapTest

```
$ git clone git@github.com:developerjohan01/TabMapTest.git`
$ cd TabMapTest/
$ ionic cordova build ios
```
TabMapTest is using a *new* project that is based on the ionic tabs starter and only have the maps plugin added... Note - no real api keys are added in the sample project

```
Saved plugin info for "cordova-plugin-ionic-keyboard" to config.xml
Discovered plugin "cordova-plugin-googlemaps" in config.xml. Adding it to the project
Installing "cordova-plugin-googlemaps" for ios
Failed to install 'cordova-plugin-googlemaps': CordovaError: Failed to fetch plugin https://github.com/mapsplugin/cordova-plugin-googlemaps-sdk/tarball/2.7.0 via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module
    at /usr/local/lib/node_modules/cordova/node_modules/cordova-lib/src/plugman/fetch.js:173:37
    at _rejected (/usr/local/lib/node_modules/cordova/node_modules/q/q.js:864:24)
    at /usr/local/lib/node_modules/cordova/node_modules/q/q.js:890:30
    at Promise.when (/usr/local/lib/node_modules/cordova/node_modules/q/q.js:1142:31)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/cordova/node_modules/q/q.js:808:41)
    at /usr/local/lib/node_modules/cordova/node_modules/q/q.js:624:44
    at runSingle (/usr/local/lib/node_modules/cordova/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/cordova/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
Failed to restore plugin "cordova-plugin-googlemaps" from config.xml. You might need to try adding it again. Error: Failed to fetch plugin https://github.com/mapsplugin/cordova-plugin-googlemaps-sdk/tarball/2.7.0 via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module
```

**Comment by wf9a5m75** [comment]
Ionic Pro does not support this plugin

**Comment by wf9a5m75** [comment]
`Failed to install 'cordova-plugin-googlemaps': CordovaError: Failed to fetch plugin https://github.com/mapsplugin/cordova-plugin-googlemaps-sdk/tarball/2.7.0 via registry.
`

https://github.com/mapsplugin/cordova-plugin-googlemaps/issues/2427

**Comment by developerjohan01** [comment]
> Ionic Pro does not support this plugin

It did until very recently...

**Comment by wf9a5m75** [comment]
I gave you a link, follow the page.

**Comment by wf9a5m75** [comment]
Then try this.

```
$> ionic cordova platform rm ios

$> ionic cordova platform add ios
```

**Comment by wf9a5m75** [comment]
As I gave proofs in https://github.com/mapsplugin/cordova-plugin-googlemaps/issues/2427, it's your problem.

I still can not detect which one is the exactly problem, but `cordova` command should be able to install the plugin from a tarball.

Let me know if you find the answer except changing the url.

**Comment by IonicProSupport** [comment]
> Ionic Pro does not support this plugin

We have released a doc detailing a workaround for Ionic Pro users. While the Ionic Pro support team doesn't officially offer support for third party plugins we will always try to provide a solution whenever possible.

[iOS Builds fail to install cordova-plugin-googlemaps](https://ionic.zendesk.com/hc/en-us/articles/360009643533-iOS-Builds-fail-to-install-cordova-plugin-googlemaps-)

This issue is reproducible locally, not just in Ionic Pro.
1- Install the plugin
2- Delete the node_modules, platforms, and plugins, dirs
3- Run npm i: npm run build; ionic cordova platform add ios
4- Note that plugin installation fails for cordova-plugin-googlemaps-sdk

**Comment by wf9a5m75** [comment]
I think this is related with ionic installer, because this problem is reported by inly ionic users. (I cannot reproduce the problem at all https://github.com/mapsplugin/cordova-plugin-googlemaps/issues/2427#issuecomment-423247760)

**Comment by wf9a5m75** [comment]
I can install the maps plugin normally  even on ubuntu (I can not develop ios app on ubuntu because there is no xcode though)

```
selfie@ubuntu:~/work$ ionic start

Every great app needs a name! 😍

Please enter the full name of your app. You can change this at any time. To bypass this prompt next time, supply name, 
the first argument to ionic start.

? Project name: testApp

Let's pick the perfect starter template! 💪

Starter templates are ready-to-go Ionic apps that come packed with everything you need to build your app. To bypass this 
prompt next time, supply template, the second argument to ionic start.

? Starter template: blank
✔ Preparing directory ./testApp - done!
✔ Downloading and extracting blank starter - done!
? Integrate your new app with Cordova to target native iOS and Android? Yes
> ionic integrations enable cordova --quiet
[INFO] Downloading integration cordova
[INFO] Copying integrations files to project
[OK] Integration cordova added!

Installing dependencies may take several minutes.

     ✨   IONIC  DEVAPP   ✨

 Speed up development with the Ionic DevApp, our fast, on-device testing mobile app

  -  🔑   Test on iOS and Android without Native SDKs
  -  🚀   LiveReload for instant style and JS updates

 -->    Install DevApp: https://bit.ly/ionic-dev-app    <--

────────────────────────────────────────────────────────────

> npm i
npm WARN deprecated browserslist@2.11.3: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
npm WARN deprecated hoek@2.16.3: The major version is no longer supported. Please update to 4.x or newer

> node-sass@4.9.0 install /home/selfie/work/testApp/node_modules/node-sass
> node scripts/install.js

Cached binary found at /home/selfie/.npm/node-sass/4.9.0/linux-ia32-57_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall /home/selfie/work/testApp/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.9.0 postinstall /home/selfie/work/testApp/node_modules/node-sass
> node scripts/build.js

Binary found at /home/selfie/work/testApp/node_modules/node-sass/vendor/linux-ia32-57/binding.node
Testing binary
Binary is fine
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"ia32"})

added 739 packages in 172.183s
> git init
Initialized empty Git repository in /home/selfie/work/testApp/.git/

     🔥   IONIC  PRO   🔥

 Supercharge your Ionic development with the Ionic Pro SDK

  -  ⚠️   Track runtime errors in real-time, back to your original TypeScript
  -  📲  Push remote updates and skip the app store queue

 Learn more about Ionic Pro: https://ionicframework.com/pro

────────────────────────────────────────────────────────────

? Install the free Ionic Pro SDK and connect your app? No
> git add -A
> git commit -m "Initial commit" --no-gpg-sign

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'selfie@ubuntu.(none)')
[WARN] Error encountered during commit. Disabling further git operations.

[INFO] Next Steps:
       
       - Go to your newly created project: cd ./testApp
       - Get Ionic DevApp for easy device testing: https://bit.ly/ionic-dev-app
selfie@ubuntu:~/work$ cd testApp/
selfie@ubuntu:~/work/testApp$ ls
config.xml  ionic.config.json  node_modules  package-lock.json  package.json  resources  src  tsconfig.json  tslint.json
selfie@ubuntu:~/work/testApp$ ionic cordova plugin add cordova-plugin-googlemaps --variable API_KEY_FOR_IOS="hoge"
✔ Creating ./www directory for you - done!
> cordova plugin add cordova-plugin-googlemaps --variable API_KEY_FOR_IOS=hoge --save

You have been opted out of telemetry. To change this, run: cordova telemetry on.
Adding cordova-plugin-googlemaps to package.json
Saved plugin info for "cordova-plugin-googlemaps" to config.xml
selfie@ubuntu:~/work/testApp$ cordova platform add ios
? May Cordova anonymously report usage statistics to improve the tool over time? No

You have been opted out of telemetry. To change this, run: cordova telemetry on.
Using cordova-fetch for cordova-ios@~4.5.4
Adding ios project...
Creating Cordova project for the iOS platform:
	Path: platforms/ios
	Package: io.ionic.starter
	Name: testApp
iOS project created with cordova-ios@4.5.5
Installing "cordova-plugin-googlemaps" for ios
apple-xcode version check failed ("/home/selfie/work/testApp/platforms/ios/cordova/apple_xcode_version"), continuing anyways.
apple-ios version check failed ("/home/selfie/work/testApp/platforms/ios/cordova/apple_ios_version"), continuing anyways.
Installing "com.googlemaps.ios" for ios

      Official document https://github.com/mapsplugin/cordova-plugin-googlemaps-doc/blob/master/v2.3.0/README.md Please consider to buy a beer for us 🍺 https://github.com/mapsplugin/cordova-plugin-googlemaps#buy-us-a-beer
    
Discovered plugin "cordova-plugin-whitelist" in config.xml. Adding it to the project
Installing "cordova-plugin-whitelist" for ios
Adding cordova-plugin-whitelist to package.json
Saved plugin info for "cordova-plugin-whitelist" to config.xml
Discovered plugin "cordova-plugin-statusbar" in config.xml. Adding it to the project
Installing "cordova-plugin-statusbar" for ios
Adding cordova-plugin-statusbar to package.json
Saved plugin info for "cordova-plugin-statusbar" to config.xml
Discovered plugin "cordova-plugin-device" in config.xml. Adding it to the project
Installing "cordova-plugin-device" for ios
Adding cordova-plugin-device to package.json
Saved plugin info for "cordova-plugin-device" to config.xml
Discovered plugin "cordova-plugin-splashscreen" in config.xml. Adding it to the project
Installing "cordova-plugin-splashscreen" for ios
Adding cordova-plugin-splashscreen to package.json
Saved plugin info for "cordova-plugin-splashscreen" to config.xml
Discovered plugin "cordova-plugin-ionic-webview" in config.xml. Adding it to the project
Installing "cordova-plugin-ionic-webview" for ios
apple-ios version check failed ("/home/selfie/work/testApp/platforms/ios/cordova/apple_ios_version"), continuing anyways.
Adding cordova-plugin-ionic-webview to package.json
Saved plugin info for "cordova-plugin-ionic-webview" to config.xml
Discovered plugin "cordova-plugin-ionic-keyboard" in config.xml. Adding it to the project
Installing "cordova-plugin-ionic-keyboard" for ios
Adding cordova-plugin-ionic-keyboard to package.json
Saved plugin info for "cordova-plugin-ionic-keyboard" to config.xml
--save flag or autosave detected
Saving ios@~4.5.5 into config.xml file ...
selfie@ubuntu:~/work/testApp$ ionic cordova plugin list
> cordova plugin ls
com.googlemaps.ios 2.7.0 "Google Maps SDK for iOS"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-googlemaps 2.4.4 "cordova-plugin-googlemaps"
cordova-plugin-ionic-keyboard 2.1.2 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 2.1.4 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
selfie@ubuntu:~/work/testApp$ ionic info
✔ Gathering environment info - done!

Ionic:

   ionic (Ionic CLI)  : 4.1.2 (/home/selfie/.nvm/versions/node/v8.11.4/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.0

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.2, cordova-plugin-ionic-webview 2.1.4, (and 5 other plugins)

System:

   NodeJS : v8.11.4 (/home/selfie/.nvm/versions/node/v8.11.4/bin/node)
   npm    : 5.6.0
   OS     : Linux 4.15

selfie@ubuntu:~/work/testApp$ 
```

**Comment by developerjohan01** [comment]
To recreate the problem, as pointed out by @IonicProSupport 

> 2- Delete the node_modules, platforms, and plugins, dirs  

Because it is _different_ to create the project from scratch compared to cloning the project from git. There are already references to the plugin when cloning the project.

This means that adding the plugin to a new project works. It also works to remove the plugin and then re-adding it.

Running `ionic cordova plugin rm cordova-plugin-googlemaps` does _not remove all_ references to googlemaps. It leaves entries in `package.json` and `com.googlemaps.ios` in the plugin directory. Remove `com.googlemaps.ios` manually and trying to add the plugin again fails. Fix this by also removing the lines
`"cordova-plugin-googlemaps": "^2.4.4",
    "cordova-plugin-googlemaps-sdk": "https://github.com/mapsplugin/cordova-plugin-googlemaps-sdk/tarball/2.7.0",`
from the `package.json`. It is now possible to add the plugin again.

My environment
```
ionic info
✔ Gathering environment info - done!

Ionic:

   ionic (Ionic CLI)  : 4.1.2 (/Users/developer/.nvm/versions/node/v10.9.0/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.0

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.2, cordova-plugin-ionic-webview 2.1.4, (and 5 other plugins)

System:

   ios-deploy : 2.0.0
   NodeJS     : v10.9.0 (/Users/developer/.nvm/versions/node/v10.9.0/bin/node)
   npm        : 6.2.0
   OS         : macOS High Sierra
   Xcode      : Xcode 10.0 Build version 10A255
```

**Comment by wf9a5m75** [comment]
Oh, I see.

> Running ionic cordova plugin rm cordova-plugin-googlemaps does not remove all references to googlemaps. It leaves entries in package.json and com.googlemaps.ios in the plugin directory. 

You mean this is a bug of ionic cli.

**Comment by developerjohan01** [comment]
No, that is _not_ what I mean - you have the same issue without using the ionic cli as far as I can tell.

``` 
cordova plugin add cordova-plugin-googlemaps \
    --variable API_KEY_FOR_ANDROID="..." \
    --variable API_KEY_FOR_IOS="..." 
```
```
cordova plugin rm cordova-plugin-googlemaps`
```

Gives the same result.

Also note that _this is just a side issue_. The _real issue is that the project does not build after cloning it from git_ or if you remove the `platforms`, and `plugins` directories - this is described in the original post and in other post. 

This started to happen after the change that is motivated in  https://github.com/mapsplugin/cordova-plugin-googlemaps/issues/2427#issuecomment-423321237
And I cant see that there is a solution in that thread.

However, there is now a [workaround for the Ionic Pro build](https://ionic.zendesk.com/hc/en-us/articles/360009643533-iOS-Builds-fail-to-install-cordova-plugin-googlemaps-) and there is also a workaround to get local builds to work by manually removing everything relating to plugin and then re-adding it.

**Comment by wf9a5m75** [comment]
As a cordova plugin developer, I expect `cordova plugin rm ` means `removing plugins including package.json`.

If `cordova` does not remove all information from files, that's a bug.
Because if you remove the plugin, but the `cordova` cli remains the plugin information in `package.json`,
it is installed even it is not used in the project.
So that's a bug of `cordova` cli.

Ask to them to fix it.

**Comment by bb128b** [comment]
Ionic's developer "workaround" is to completely destroy the package-lock.json file. That's not a stable workaround in my opinion as the lock file dictates that our CI build packages do not go past a certain version to ensure no breakages with our RC.

In order to fix this issue I had to go through all my Ionic Native packages (mainly Google Maps) and update every one and check over the app to ensure nothing broke.

**Comment by developerjohan01** [comment]
@bb128b - i feel your pain - the Ionic pro workaround is just a _workaround_ for a different problem... However, now the build is broken again because of the introduction of a new dependency to cocapods see https://github.com/mapsplugin/cordova-plugin-googlemaps/issues/2434

**Comment by bb128b** [comment]
Just to clarify. All issues can be resolved (temporarily) by "pinning" your Google maps cordova plugin to 2.4.4.

```
"cordova-plugin-googlemaps": "2.4.4",
```

At least until the dust is settled on this "major" change.

**Comment by developerjohan01** [comment]
The way I sorted my local build was to manually remove the plugin and then first install _cocapods_ - the new dependency introduced by the plugin. 
I already got [homebrew](https://brew.sh/) installed so I use that. 
```
brew update
brew install cocoapods
pod setup
```  
I then re-installed the plugin.
And I can now build the project.
I have also removed any workaround previously applied. e.g. I have re-enabled the `package-lock.json` file.
For the **Ionic Pro** build to work I use the following workaround:
https://ionic.zendesk.com/hc/en-us/articles/360001581893-iOS-build-fails-to-fetch-onesignal-cordova-plugin
The build server obviously need to have cocapods installed.
